includes/class-salesforce.php

```diff
@@ -28,6 +28,133 @@ class Salesforce {
 		'webhook_id'    => '',
 	];
 
+	/**
+	 * Add hooks.
+	 */
+	public static function init() {
+		add_action( 'rest_api_init', [ __CLASS__, 'register_api_endpoints' ] );
+	}
+
+	/**
+	 * Register the endpoints needed to handle Salesforce sync webhooks.
+	 */
+	public static function register_api_endpoints() {
+		// Handle WooCommerce webhook to sync with Salesforce.
+		register_rest_route(
+			NEWSPACK_API_NAMESPACE,
+			'/salesforce/sync',
+			[
+				'methods'             => \WP_REST_Server::EDITABLE,
+				'callback'            => [ __CLASS__, 'api_sync_salesforce' ],
+				'permission_callback' => '__return_true',
+			]
+		);
+	}
+
+	/**
+	 * Webhook callback handler for syncing data to Salesforce.
+	 *
+	 * @param WP_REST_Request $request Request containing webhook.
+	 * @return WP_REST_Response|WP_Error The response from Salesforce, or WP_Error.
+	 */
+	public static function api_sync_salesforce( $request ) {
+		$args          = $request->get_params();
+		$order_details = self::parse_wc_order_data( $args );
+
+		if ( empty( $order_details ) ) {
+			return new \WP_Error(
```

Comment:
Errors should also be returned with `\rest_ensure_response();`

Reply:
Good call, updated in 6d4574c4.